pulp_container/app/models.py

```diff
@@ -367,57 +367,52 @@ def generate_filename(instance, filename):
 
 class Upload(BaseModel):
     """
-    Model for tracking Blob uploads.
+    A model for tracking Blob uploads.
     """
 
     repository = models.ForeignKey(Repository, related_name="uploads", on_delete=models.CASCADE)
+    cumulative_size = models.BigIntegerField(default=0)
+
+
+class BlobTemporaryUpload(PulpTemporaryFile):
+    """
+    A model used for storing uploaded blob chunks in a temporary file.
+    """
```

Comment:
Would it make sense (and maybe the code easier) if you didn't inherit from `PulpTemporaryFile`, but have a one-to-one relation with cascaded delete here?
It would render this model to be a rich (with the additional `offset`) join table from `Upload` to `PulpTemporaryFile`.